src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -161,10 +161,15 @@ public SolutionState WithNewWorkspace(Workspace workspace, int workspaceVersion)
         // [Conditional("DEBUG")]
```

Comment:
Want to re-activate this for DEBUG only? SetEquals isn't cheap.

Reply:
(or at least, I expect it'll allocate a lot more than the allocations you're trying to save in the ProjectDependencyGraph code!)

Reply:
I'm using `Debug.Assert` for those, not `Contract` for that reason.

Reply:
Oh right those are Debug.Assert and not Contract APIs...point might still stand.

Reply:
The compiler does not emit arguments of `Debug.Assert` in release build.